version: '2.1'
services:
  # Application
  web:
    extends:
      file: docker-services.yml
      service: app
    command: zenodo run -h 0.0.0.0 -p 5000
    environment:
      - "INVENIO_SESSION_COOKIE_SECURE=False"
      - "INVENIO_STATSD_HOST="
      - VIRTUAL_HOST=zenodo.ubuntunet.net
      - VIRTUAL_PORT=5000
      - LETSENCRYPT_EMAIL=repository@ubuntunet.net
      - LETSENCRYPT_HOST=zenodo.ubuntunet.net
    ports:
      - "5000:5000"
    links:
      - cache
      - es
      - mq
      - db
    volumes_from:
      - static
    networks:
      - zenodo-net
      - nginx-proxy
  worker:
    extends:
      file: docker-services.yml
      service: app
    restart: "always"
    command: "celery worker -A zenodo.celery --loglevel=INFO"
    links:
      - cache
      - es
      - mq
      - db
    volumes_from:
      - static
    networks:
      - zenodo-net
  static:
    extends:
      file: docker-services.yml
      service: app
    restart: "no"
    command: tail -f /var/log/faillog
    user: zenodo
    volumes:
      - "/usr/local/var/instance/static"
      - "/usr/local/lib/python3.5/site-packages"
      - "/usr/local/var/data"
      - ".:/code/zenodo"
    networks:
      - zenodo-net

  # Monitoring
  flower:
    extends:
      file: docker-services.yml
      service: flower
    links:
      - mq
    networks:
      - zenodo-net

  # Base services
  cache:
    extends:
      file: docker-services.yml
      service: cache
    networks:
      - zenodo-net
  db:
    extends:
      file: docker-services.yml
      service: db
    networks:
      - zenodo-net
  mq:
    extends:
      file: docker-services.yml
      service: mq
    networks:
      - zenodo-net
  es:
    extends:
      file: docker-services.yml
      service: es
    networks:
      - zenodo-net
  kibana:
    extends:
      file: docker-services.yml
      service: kibana
    links:
      - es
    networks:
      - zenodo-net

networks:
  zenodo-net:
  nginx-proxy:
    external: true
